File: backend-api/app/domain/whatsapp_account.py
"""Domain models for tenant WhatsApp accounts."""
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class WhatsAppAccount(BaseModel):
    account_id: str
    tenant_id: str
    account_key: str
    display_name: str
    phone_number_id: str
    display_phone_number: str
    verify_token: str | None = None
    status: str = "active"
    is_default: bool = False
    has_access_token: bool = False
    masked_access_token: str | None = None
    created_at: datetime | None = None
    updated_at: datetime | None = None


class WhatsAppAccountSecret(BaseModel):
    account: WhatsAppAccount
    access_token: str | None = None


class WhatsAppAccountCreate(BaseModel):
    account_key: str
    display_name: str
    phone_number_id: str
    display_phone_number: str
    access_token: str | None = None
    verify_token: str | None = None
    status: str = "active"
    is_default: bool = False


class WhatsAppAccountUpdate(BaseModel):
    display_name: str | None = None
    phone_number_id: str | None = None
    display_phone_number: str | None = None
    access_token: str | None = None
    verify_token: str | None = None
    status: str | None = None
    is_default: bool | None = None

File: admin-panel/lib/models/whatsapp_account.dart
class WhatsAppAccountModel {
  WhatsAppAccountModel({
    required this.accountId,
    required this.tenantId,
    required this.accountKey,
    required this.displayName,
    required this.phoneNumberId,
    required this.displayPhoneNumber,
    required this.status,
    required this.isDefault,
    required this.hasAccessToken,
    this.maskedAccessToken,
    this.verifyToken,
  });

  final String accountId;
  final String tenantId;
  final String accountKey;
  final String displayName;
  final String phoneNumberId;
  final String displayPhoneNumber;
  final String status;
  final bool isDefault;
  final bool hasAccessToken;
  final String? maskedAccessToken;
  final String? verifyToken;

  factory WhatsAppAccountModel.fromJson(Map<String, dynamic> json) {
    return WhatsAppAccountModel(
      accountId: json['account_id']?.toString() ?? '',
      tenantId: json['tenant_id']?.toString() ?? '',
      accountKey: json['account_key']?.toString() ?? '',
      displayName: json['display_name']?.toString() ?? '',
      phoneNumberId: json['phone_number_id']?.toString() ?? '',
      displayPhoneNumber: json['display_phone_number']?.toString() ?? '',
      status: json['status']?.toString() ?? 'active',
      isDefault: json['is_default'] == true,
      hasAccessToken: json['has_access_token'] == true,
      maskedAccessToken: json['masked_access_token']?.toString(),
      verifyToken: json['verify_token']?.toString(),
    );
  }
}

